File: TransCoder_tokenized_test_set_functions/cpp/PAPER_CUT_MINIMUM_NUMBER_SQUARES_SET_2.cpp
int minimumSquare ( int m, int n ) {
  int vertical_min = INT_MAX;
  int horizontal_min = INT_MAX;
  if ( m == n ) return 1;
  if ( dp [ m ] [ n ] ) return dp [ m ] [ n ];
  for ( int i = 1;
  i <= m / 2;
  i ++ ) {
    horizontal_min = min ( minimumSquare ( i, n ) + minimumSquare ( m - i, n ), horizontal_min );
  }
  for ( int j = 1;
  j <= n / 2;
  j ++ ) {
    vertical_min = min ( minimumSquare ( m, j ) + minimumSquare ( m, n - j ), vertical_min );
  }
  dp [ m ] [ n ] = min ( vertical_min, horizontal_min );
  return dp [ m ] [ n ];
}

File: TransCoder_tokenized_test_set_functions/python/PAPER_CUT_MINIMUM_NUMBER_SQUARES_SET_2.py
def minimumSquare ( m, n ) :
    vertical_min = 10000000000
    horizontal_min = 10000000000
    if m == n :
        return 1
    if dp [ m ] [ n ] != 0 :
        return dp [ m ] [ n ]
    for i in range ( 1, m //2+1):
        horizontal_min=min(minimumSquare(i,n)+minimumSquare(m-i,n),horizontal_min)
    forjinrange(1,n//2+1):
        vertical_min=min(minimumSquare(m,j)+minimumSquare(m,n-j),vertical_min)
    dp[m][n]=min(vertical_min,horizontal_min)
    returndp[m][n]
